internal/benchrunner/runners/rally/runner.go

```diff
@@ -961,14 +1072,17 @@ func generateRallyTrack(dataStream string, corpusFile *os.File, corpusDocsCount
 
 	corpusSizeInBytes := fi.Size()
 
-	buf := new(bytes.Buffer)
 	templateData := map[string]any{
-		"DataStream":        dataStream,
-		"CorpusFilename":    filepath.Base(corpusFile.Name()),
-		"CorpusDocsCount":   corpusDocsCount,
-		"CorpusSizeInBytes": corpusSizeInBytes,
+		"DataStream":         dataStream,
+		"CorpusFilename":     filepath.Base(corpusFile.Name()),
+		"CorpusDocsCount":    corpusDocsCount,
+		"CorpusSizeInBytes":  corpusSizeInBytes,
+		"ComposableTemplate": dataStream,
```

Comment:
I expected this to be call `@custom` or similar. Have not run the code on my end yet, can you share what the exact name of this composable template is that is loaded?

Reply:
it is something like: `metrics-kubernetes.pod-ep` (the same name as the data stream)

`metrics-kubernetes.pod@custom` is managed by the integration package, and it was not enough to there the `time_series.start_time/end_time` and `routing_path`, not sure why.

I ended up calling the simulate index template APi for `metrics-kubernetes.pod` and generate a copy of the template just with different `time_series.start_time/end_time` and higher priority, to apply to index pattern `metrics-kubernetes.pod-ep`

Reply:
I'm worried we modify here the package instead of just extending it. `metrics-kubernetes.pod@custom` is not managed by the integration package, it is only referenced and the goal is to use it exactly for customisations. 

Will read up on https://github.com/elastic/integrations/issues/8431#issuecomment-1807882262 to better understand what the issue there is.

Reply:
In case we move forward with the proposed approach, could this have any sideffects? For example, the same dataset is used with newer versions of the package? So far, this meant the newer package and all the new templates are installed. But with the solution we have here it means always the old "templates" are installed. Or does it still mostly work, as all the "important" bits are in the components which are referenced?

Reply:
> I'm worried we modify here the package instead of just extending it. `metrics-kubernetes.pod@custom` is not managed by the integration package, it is only referenced and the goal is to use it exactly for customisations.

I looked in the Kibana UI and `metrics-kubernetes.pod@custom` is marked as a managed template.
I will try another time to add the minimum settings required for overriding `time_series.start_time/end_time` in `metrics-kubernetes.pod@custom` and try to understand why it does not work.
From my point of view we are not exactly modifying the package, rather overriding for `metrics-kubernetes.pod-ep` the templates from the package with another separated template that's a copy/merge of all the ones from the package.



> In case we move forward with the proposed approach, could this have any sideffects? For example, the same dataset is used with newer versions of the package? So far, this meant the newer package and all the new templates are installed. But with the solution we have here it means always the old "templates" are installed. Or does it still mostly work, as all the "important" bits are in the components which are referenced?

In case of a new version of the package, the template in the rally track might be obsolete. You should regenerate the track in order to be up to date. If you run the benchmark directly from `elastic-package` the template will be always up to date with the package in the local folder we install as zip.

Note that the above will be true even if we change only `metrics-kubernetes.pod@custom`, as long as soon we have a new package with changes in that template.

I'm indeed a little confused by how `metrics-kubernetes.pod@custom` is supposed to work.
What if we run `elastic-package benchmark rally` on cluster where `metrics-kubernetes.pod@custom` was already (well :)) customized?

with the current solution we will keep the customization in the overriding template.
If we use `metrics-kubernetes.pod@custom` I think we should to the same, adding the settings for time_series.start_time/end_time` and finally cleaning it.
Let me add that cleaning up an "ephemeral" template like the one in the current solution might be easier to mantain that cleaning up `metrics-kubernetes.pod@custom` :)

Reply:
@ruflin Indeed I've realized that if we create an index template with the same content in the rally track, we can replay the track directly with rally even without installing the package.
if we use `metrics-kubernetes.pod@custom` before replaying the rally track we should also install the package, that would anyway be true since we'd loose also ingestion pipelines

Reply:
> `metrics-kubernetes.pod@custom` is not managed by the integration package

```
{"component_templates":[{"name":"metrics-kubernetes.pod@custom","component_template":{"template":{"settings":{}},"_meta":{"package":{"name":"kubernetes"},"managed_by":"fleet","managed":true}}}]}
```

Reply:
That `@custom` is marked as managed but is not seems to have some confusion recently. The is an optional template, meaning it is only used if it is there. @kpollich Pinging you here as the confusion around `@custom` has popped up in other palces too.

> @ruflin Indeed I've realized that if we create an index template with the same content in the rally track, we can replay the track directly with rally even without installing the package.

I would assume the template you export is only the one that links to all the component templates, so you still needed the component templates from the package?

Reply:
Thanks for the ping, @ruflin. Taking a look at the [code path](https://github.com/elastic/kibana/blob/2a8a68c1f8d14074d23ca9feef118aa172191bfe/x-pack/plugins/fleet/server/services/epm/elasticsearch/template/install.ts#L400C19-L406) in Fleet, we never removed the logic to create an empty `@custom` component template here. Since `ignore_missing_templates` exists now we should probably remove this. 

I created https://github.com/elastic/kibana/issues/171540 to capture the removal.

Reply:
> I would assume the template you export is only the one that links to all the component templates, so you still needed the component templates from the package?

I simulate the template `metrics-kubernetes.pod` template, that's composed of all the templates from the package, ending up with a single index temple body that's equivalent to the one that would be used for `metrics-kubernetes.pod-ep` data stream if the package would be installed. it will indeed bring as well any customisation made in `metrics-kubernetes.pod@custom` at the time of the simulation, if any. that's the reason why I don't need the component template from the package.

The simulate API is used to "expand" any implicit of default setting, like indeed the one for `index.routing_path` :)

I've pushed a commit with customising instead (well) `metrics-kubernetes.pod@custom`: I still need to run the simulate API for the reason just above.

The clean up in this case will update back `metrics-kubernetes.pod@custom` with the content before it had increasing the time series start/end time slot.

beware that this approach does not work because a validation issue from ES that complains about an empty `index.routing_path` (that's not) on the final merged `metrics-kubernetes.pod` template

Reply:
How does the discussion we had yesterday around exporting "all" Elasticsearch assets to the rally track change this discussion?

Reply:
if we export "all" ES assets, we must have a way to regenerate the track without overwriting the corpus, in order to "dump" to a different version of the package

in this case we will export everything and so we can export as well the `metrics-kubernetes.pod@custom` with the custom (well) `time_series.start_date/end_date`

I would rather suggest, at light of the discussion we had yesterday, that we install the package separately, either directly with `elastic-package` (either by the binary itself, or orchestrating `esbench`) having on option to re-play an existing rally track

Reply:
Lets go through the scenarios:

* Running a quick benchmark directly from elastic-package. The approach in this PR works. The ideal would be with @custom but this is not work in ES
* Create rally track with all assets: This template + export of all ingest pipelines which then are loaded by rally would get the job done. The rally track then is for a specific version of the package. We need to find ways to generate the corpus separately
* Corpus only generation: Is this a new command / flag we need?

What I don't like about the above is that this should also work for `stream`, meaning we adjust the date. What is the index template / component template where the time range change could be made and would work today taking the `metrics-aws.sqs` dataset?

Long term goal is that package installation happens through esbench / elastic-package and only thing the rally track needs is to reference a package and it's custom additional bits but we are too far away from this at the moment.